src/navigation/context.py

```diff
@@ -29,18 +30,54 @@ class Gate:
 @dataclass
 class FailureZone:
     """
-    FailureZones are represented as rectangles
-    vertices should be a np.ndarray of shape (4, 2) representing 
-    2D corners of the failure zone in clockwise order 
+    FailureZones are represented as rectangles.
+    self.vertices should be a np.ndarray of shape (4, 2) representing 
+    the 4 2D corners of the failure zone such that v0, v2 are diagonal
     """
     vertices: np.ndarray    # shape (4, 2)
 
-    def intersect(self, path_vec: np.ndarray) -> bool:
+    def intersect(self, path_start: np.array, path_end: np.array) -> bool:
```

Comment:
in the gate path pr, we're using shapely (https://shapely.readthedocs.io/en/stable/manual.html) to just do this for us, we should probably be consistent. Do you see a good reason to implement these functions ourselves as opposed to just using the ones provided by the library?

Reply:
Yeah I saw that after I implemented this stuff, I think that would work well. It would probably add overhead + be slower than these (because these functions are implemented very specifically for this algorithm), but I doubt it would matter much. I'll change to use those for now (and we can always put this back in if it's slow).